guides/plugins/plugins/content/seo/add-custom-seo-url.md

```diff
@@ -0,0 +1,356 @@
+parameter for the `importTranslation` method.
+
+Let's have a look at an example:
+
+{% code title="<plugin root>/src/Migration/Migration1619094740AddStaticSeoUrl.php" %}
+```php
+<?php declare(strict_types=1);
+
+namespace Swag\BasicExample\Migration;
+
+use Doctrine\DBAL\Connection;
+use Shopware\Core\Defaults;
+use Shopware\Core\Framework\Migration\MigrationStep;
+use Shopware\Core\Framework\Uuid\Uuid;
+use Shopware\Core\Migration\Traits\ImportTranslationsTrait;
+use Shopware\Core\Migration\Traits\Translations;
+
+class Migration1619094740AddStaticSeoUrl extends MigrationStep
+{
+    use ImportTranslationsTrait;
+
+    public function getCreationTimestamp(): int
+    {
+        return 1619094740;
+    }
+
+    public function update(Connection $connection): void
+    {
+        $this->importTranslation('seo_url', new Translations(
+            // German array
+            array_merge($this->getSeoMetaArray($connection), ['seo_path_info' => 'Beispiel-Seite']),
+            // English array
+            array_merge($this->getSeoMetaArray($connection), ['seo_path_info' => 'Example-Page']),
+
+        ), $connection);
+    }
+
+    public function updateDestructive(Connection $connection): void
+    {
+    }
+
+    private function getSeoMetaArray(Connection $connection): array
+    {
+        return [
+            'id' => Uuid::randomBytes(),
+            'sales_channel_id' => $this->getStorefrontSalesChannelId($connection),
+            'foreign_key' => Uuid::randomBytes(),
+            'route_name' => 'frontend.example.example',
+            'path_info' => '/example',
+            'is_canonical' => 1,
+            'is_modified' => 0,
+            'is_deleted' => 0,
+        ];
+    }
+
+    private function getStorefrontSalesChannelId(Connection $connection): ?string
+    {
+        $sql = <<<SQL
+            SELECT id
+            FROM sales_channel
+            WHERE type_id = :typeId
+SQL;
+        $salesChannelId = $connection->fetchOne($sql, [
+            ':typeId' => Uuid::fromHexToBytes(Defaults::SALES_CHANNEL_TYPE_STOREFRONT)
+        ]);
+
+        if (!$salesChannelId) {
+            return null;
+        }
+
+        return $salesChannelId;
+    }
+}
+```
+{% endcode %}
+
+You might want to have a look at the `getSeoMetaArray` method, that we implemented here.
+Most important for you are the columns `route_name` and `path_info` here, which represent the values you've defined in
+your controller's route annotation.
+
+By using the default PHP method `array_merge`, we're then also adding our translated SEO URL to the column `seo_path_info`.
+
+And that's it! After installing our plugin, you should now be able to access your controller's route with the given SEO URLs.
+
+{% hint style="info"%}
+You can only access the German SEO URL if you've configured a German domain in your respective sales channel first.
+{% endhint %}
+
+### Dynamic SEO URLs
+
+Dynamic SEO URLs are URLs, that have to change every now and then.
```

Comment:
Change this to non-entity examples. Use e.g. `ProductPageSeoUrlRoute` instead for entity examples

Reply:
Also add: Must execute `SeoUrlUpdater::update` on written event still when using the PageSeourlRoute class